File: KoRadio/KoRadio.Services/Database/UserRole.cs
using System;
using System.Collections.Generic;

namespace KoRadio.Services.Database;

public partial class UserRole
{
    public int UserRolesId { get; set; }

    public int? UserId { get; set; }

    public int? RoleId { get; set; }

    public DateTime? ChangedAt { get; set; }
    public User User { get; set; }
    public Role Role { get; set; }
}

File: KoRadio/UI/ko_radio_desktop/lib/models/user_role.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:ko_radio_desktop/models/role.dart';
part 'user_role.g.dart';
@JsonSerializable()
class UserRole {
	int userRoleId;
	int userId;
	int roleId;
  DateTime createdAt;
	DateTime changedAt;
	Role role;

  UserRole({ required this.userRoleId, required this.userId, required this.roleId, required this.createdAt, required this.changedAt, required this.role });

	factory UserRole.fromJson(Map<String, dynamic> json) => _$UserRoleFromJson(json);

	Map<String, dynamic> toJson() => _$UserRoleToJson(this);
}